### ISSUE TITLE
Different results from FindMarkers by Seurat Version 5

### ISSUE THREAD

**Issue by Newbie-Anastasia** [satijalab/seurat#8042]
Hello Seurat team,

I have been cruising around trying to find an answer to my current bug without success.
I have recently updated seurat to v5 and while rerunning my code while all functions seem to run smoothly the Findmarkers() and findallmarkers() function don't (with the exception when I use the "roc" test): The calculation that previously took 20/30min now does not even show a calculation bar and runs in 30secondes. Giving at the end a fully pink heat map with little to no gene expression.
I have tried to change the pseudocount values from the default to 0.1 (as it has been apparently changed it in the new version) without success.
I am wondering what other parameters have been changed that could affect these functions and prevent their calculations?

Thanks for any possible help!

**Comment by mhkowalski** [comment]
Hi, 

Thanks for your question. We did change the `FindMarkers()` calculation, as described [here](https://satijalab.org/seurat/articles/announcements). We now use presto to calculate `FindMarkers`, which explains the speedup in time you observe (and lack of calculation bar). 

As we describe above, lowly expressed genes may have a higher log2FC due to changes in the pseudocout. We currently rank the results first by p-value, and then by log2FC. Thus, the pink heatmap is likely due to lowly expressed genes that now I have a high log2FC (in addition to low p-values). To get around this, a better way of sorting is by p-value, and then by the absolute difference in percent1 and percent2. We are currently updating Seurat to sort the `FindMarkers` results in this manner, but in the meantime you could sort first by p-value, and then by  decreasing `abs_pct_diff`, as shown below.

```
library(dplyr)
de_results$abs_pct_diff <- abs(de_results$pct.1 - de_results$pct.2)
de_results %>% arrange(p_val_adj, desc(abs_pct_diff)) 
```

**Comment by Newbie-Anastasia** [comment]
Hello!

Thank you for your response I will try it with your advice!

**Comment by Newbie-Anastasia** [comment]
Still not working, it does reorder the heat map but it still looks abnormal.
Here without: de_results$abs_pct_diff <- abs(de_results$pct.1 - de_results$pct.2)
de_results %>% arrange(p_val_adj, desc(abs_pct_diff)) 
![Screenshot 2023-11-20 at 4 43 29 PM](https://github.com/satijalab/seurat/assets/151047318/df89a066-7b89-4ee4-a90a-e5ca5f61df8c)
Here with:

![Screenshot 2023-11-20 at 4 44 35 PM](https://github.com/satijalab/seurat/assets/151047318/1b8d8fca-5f53-4684-8884-d00ce4c1c2eb)

Compared to previously, an example here:
![Screenshot 2023-11-20 at 4 45 45 PM](https://github.com/satijalab/seurat/assets/151047318/4d71a3fb-73cb-47e5-aa0a-3221e33fb448)

**Comment by rsatija** [comment]
This was fixed quickly after the initial release in 5.0 (5.0.1). We no longer sort genes with tied p-values by logFC, since the logFC can be very large and unstable for lowly expressed genes.

Please re-install on CRAN and rerun FindMarkers - if this is still an issue please re-open and we will debug
